yb-voyager/cmd/common.go

```diff
@@ -1176,20 +1176,36 @@ func getMigrationComplexityForOracle(schemaDirectory string) (string, error) {
-	TargetDBVersion            *ybversion.YBVersion                  `json:"TargetDBVersion"`
-	MigrationComplexity        string                                `json:"MigrationComplexity"`
-	SchemaSummary              utils.SchemaSummary                   `json:"SchemaSummary"`
-	Sizing                     *migassessment.SizingAssessmentReport `json:"Sizing"`
-	UnsupportedDataTypes       []utils.TableColumnsDataTypes         `json:"UnsupportedDataTypes"`
-	UnsupportedDataTypesDesc   string                                `json:"UnsupportedDataTypesDesc"`
-	UnsupportedFeatures        []UnsupportedFeature                  `json:"UnsupportedFeatures"`
-	UnsupportedFeaturesDesc    string                                `json:"UnsupportedFeaturesDesc"`
-	UnsupportedQueryConstructs []utils.UnsupportedQueryConstruct     `json:"UnsupportedQueryConstructs"`
-	UnsupportedPlPgSqlObjects  []UnsupportedFeature                  `json:"UnsupportedPlPgSqlObjects"`
-	MigrationCaveats           []UnsupportedFeature                  `json:"MigrationCaveats"`
-	TableIndexStats            *[]migassessment.TableIndexStats      `json:"TableIndexStats"`
-	Notes                      []string                              `json:"Notes"`
+	VoyagerVersion      string                                `json:"VoyagerVersion"`
+	TargetDBVersion     *ybversion.YBVersion                  `json:"TargetDBVersion"`
+	MigrationComplexity string                                `json:"MigrationComplexity"`
+	SchemaSummary       utils.SchemaSummary                   `json:"SchemaSummary"`
+	Sizing              *migassessment.SizingAssessmentReport `json:"Sizing"`
+	Issues              []AssessmentIssue                     `json:"-"` // disabled in reports till corresponding UI changes are done(json and html reports)
+	TableIndexStats     *[]migassessment.TableIndexStats      `json:"TableIndexStats"`
+	Notes               []string                              `json:"Notes"`
+
+	// fields going to be deprecated
+	UnsupportedDataTypes       []utils.TableColumnsDataTypes     `json:"UnsupportedDataTypes"`
+	UnsupportedDataTypesDesc   string                            `json:"UnsupportedDataTypesDesc"`
+	UnsupportedFeatures        []UnsupportedFeature              `json:"UnsupportedFeatures"`
+	UnsupportedFeaturesDesc    string                            `json:"UnsupportedFeaturesDesc"`
+	UnsupportedQueryConstructs []utils.UnsupportedQueryConstruct `json:"UnsupportedQueryConstructs"`
+	UnsupportedPlPgSqlObjects  []UnsupportedFeature              `json:"UnsupportedPlPgSqlObjects"`
+	MigrationCaveats           []UnsupportedFeature              `json:"MigrationCaveats"`
+}
+
+type AssessmentIssue struct {
+	Category              string
+	CategoryDescription   string
+	TypeName              string
+	TypeDescription       string
+	Impact                string
+	ObjectType            string
+	ObjectName            string
+	SqlStatement          string
```

Comment:
we can still use the `ObjectInfo` for these three fields ?

Reply:
but goal is keep the new AssessmentIssue struct flat if possible.
Planning to remove ObjectInfo struct once we move away from the current struct format.